src/test/java/org/jnode/fs/ntfs/NtfsReparsePointTest.java

```diff
@@ -29,9 +30,12 @@ public void testResidentReparsePoint()  {
                         " 0 49 0 50 0 51 0 52 0 53 0 54 0 55 0 56 0 57 0 92 0 48 0 49 0 50 0 51 0 52 0 53 0 54 0 55 0 56" +
                         " 0 57 0 92 0 48 0 49 0 50 0 51 0 52 0 53 0 54 0 55 0 56 0 57 0 92 0 48 0 49 0 50 0 51 0 0 0 0 0 0 0");
 
-        ReparsePointAttribute reparsePointAttribute = new ReparsePointAttribute(new NTFSStructure(data, 0), 0);
+        ReparsePointAttributeRes reparsePointAttributeRes = new ReparsePointAttributeRes(new NTFSStructure(data, 0), 0);
 
-        assertThat(reparsePointAttribute.getPrintName(), is ("c:\\temp\\0123456789\\0123456789\\0123456789\\0123456789\\0123456789\\0123456789\\0123456789\\0123456789\\0123"));
-        assertThat(reparsePointAttribute.getTargetName(), is ("\\??\\c:\\temp\\0123456789\\0123456789\\0123456789\\0123456789\\0123456789\\0123456789\\0123456789\\0123456789\\0123"));
+        assertThat(reparsePointAttributeRes.getReparseTag(), is(equalTo(ReparsePointTags.IO_REPARSE_TAG_MOUNT_POINT)));
+        assertThat(reparsePointAttributeRes.getReparseDataLength(), is(equalTo(420)));
+
+        // print name: "c:\\temp\\0123456789\\0123456789\\0123456789\\0123456789\\0123456789\\0123456789\\0123456789\\0123456789\\0123"));
```

Comment:
why can't them be verified now?

Reply:
I'd incorrectly assumed that the type was always the same. There's actuallyI'd incorrectly assumed that the type was always the same many different data types stored in the reparse attribute, so I've moved the type specific parsing out of jnode-fs.